tests/common.go

```diff
@@ -358,6 +360,54 @@ func ValidateAndDestroy(contexts []*scheduler.Context, opts map[string]bool) {
 	})
 }
 
+// AddLabelsOnNode adds labels on the node
+func AddLabelsOnNode(n node.Node, labels map[string]string) error {
+	for labelKey, labelValue := range labels {
+		if err := Inst().S.AddLabelOnNode(n, labelKey, labelValue); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ValidateStoragePools validates sizes of storage pools based on autopilot rules
+func ValidateStoragePools(contexts []*scheduler.Context, apRules []ap_api.AutopilotRule) {
+	var err error
+	var expectedStoragePoolsWorkloadSize = make(map[string]uint64)
+	for _, ctx := range contexts {
+		Step(fmt.Sprintf("get replica sets %s app volumes", ctx.App.Key), func() {
+			appVolumes, err := Inst().S.GetVolumes(ctx)
+			expect(err).NotTo(haveOccurred())
+			expect(appVolumes).NotTo(beEmpty())
+			for _, volume := range appVolumes {
+				if Inst().S.IsAutopilotEnabledForPvc(volume.Annotations) {
```

Comment:
The IsAutopilotEnabledForPvc() interface method should not be needed.

The logic below should upgrade the `.StoragePools[idx].WorkloadSize` regardless of whether autopilot is enabled or not. The fact that a workload runs on a storage pool doesn't have direct correlation to autopilot running in the cluster.

Now when we do `ValidateStoragePoolsSizes`, that's where the volume driver can detect if autopilot is enabled for that pool and then check if the pool expanded as per the workload size. If autopilot is not enabled, then the pool is expected to stay at it's original size and the validation should pass.

In general, I'm trying to reduce the many if conditionals we have here and the indirect/implicit expectations between different code pieces.